src/uct/ib/rc/accel/gga_mlx5.c

```diff
@@ -185,17 +185,121 @@ uct_gga_mlx5_rkey_resolve(uct_ib_mlx5_md_t *md, uct_rkey_t rkey)
+};
+
+static uct_rc_iface_ops_t uct_gga_mlx5_iface_ops = {
+    .super = {
+        .super = {
+            .iface_estimate_perf   = (uct_iface_estimate_perf_func_t)ucs_empty_function_do_assert,
+            .iface_vfs_refresh     = uct_rc_iface_vfs_refresh,
+            .ep_query              = (uct_ep_query_func_t)ucs_empty_function,
+            .ep_invalidate         = uct_rc_mlx5_base_ep_invalidate,
+            .ep_connect_to_ep_v2   = (uct_ep_connect_to_ep_v2_func_t)ucs_empty_function_do_assert,
+            .iface_is_reachable_v2 = ucs_empty_function_do_assert,
+            .ep_is_connected       = ucs_empty_function_do_assert
+        },
+        .create_cq      = uct_rc_mlx5_iface_common_create_cq,
+        .destroy_cq     = uct_rc_mlx5_iface_common_destroy_cq,
+        .event_cq       = uct_rc_mlx5_iface_common_event_cq,
+        .handle_failure = (uct_ib_iface_handle_failure_func_t)ucs_empty_function_do_assert_void,
+    },
+    .init_rx         = (uct_rc_iface_init_rx_func_t)ucs_empty_function_do_assert,
+    .cleanup_rx      = ucs_empty_function_do_assert_void,
+    .fc_ctrl         = uct_rc_mlx5_base_ep_fc_ctrl,
+    .fc_handler      = uct_rc_iface_fc_handler,
+    .cleanup_qp      = ucs_empty_function_do_assert_void,
+    .ep_post_check   = uct_rc_mlx5_base_ep_post_check,
+    .ep_vfs_populate = uct_rc_mlx5_base_ep_vfs_populate
+};
+
 static UCS_CLASS_INIT_FUNC(uct_gga_mlx5_iface_t,
                            uct_md_h tl_md, uct_worker_h worker,
                            const uct_iface_params_t *params,
                            const uct_iface_config_t *tl_config)
 {
-    return UCS_ERR_NOT_IMPLEMENTED;
+    uct_gga_mlx5_iface_config_t *config =
+            ucs_derived_of(tl_config, uct_gga_mlx5_iface_config_t);
+    uct_ib_mlx5_md_t *md                = ucs_derived_of(tl_md, uct_ib_mlx5_md_t);
+    uct_ib_iface_init_attr_t init_attr  = {};
+
+    if (!(md->flags & UCT_IB_MLX5_MD_FLAG_MMO_DMA)) {
+        return UCS_ERR_UNSUPPORTED;
+    }
+
+    init_attr.fc_req_size           = sizeof(uct_rc_pending_req_t);
+    init_attr.flags                 = UCT_IB_CQ_IGNORE_OVERRUN | UCT_IB_IS_GGA;
+    init_attr.cq_len[UCT_IB_DIR_TX] = config->super.tx_cq_len;
+    init_attr.qp_type               = IBV_QPT_RC;
+    init_attr.max_rd_atomic         = 0;
+
+    uct_ib_mlx5_parse_cqe_zipping(md, &config->rc_mlx5_common.super,
+                                  &init_attr);
+
+    UCS_CLASS_CALL_SUPER_INIT(uct_rc_mlx5_iface_common_t,
+                              &uct_gga_mlx5_iface_tl_ops,
+                              &uct_gga_mlx5_iface_ops, tl_md, worker, params,
+                              &config->super.super, &config->rc_mlx5_common,
+                              &init_attr);
+
+    self->super.super.config.tx_moderation =
+            ucs_min(config->super.tx_cq_moderation, self->super.tx.bb_max / 4);
+    config->super.super.fc.enable          = 0;
+    return uct_rc_init_fc_thresh(&config->super, &self->super.super);
```

Comment:
it's code duplication with uct_rc_mlx5_iface_t constructor

Reply:
can we remove calling `uct_rc_init_fc_thresh(&config->super, &self->super.super);` from gga?